scripts/determine-reboot-cause

```diff
@@ -223,7 +223,7 @@ def main():
         os.makedirs(REBOOT_CAUSE_DIR)
 
     # Remove stale PREVIOUS_REBOOT_CAUSE_FILE if it exists
-    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE):
+    if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE) or os.path.islink(PREVIOUS_REBOOT_CAUSE_FILE):
```

Comment:
> @mlok-nokia can you please bring this check to the line 249 where symlink file is created? so that its quite clear

@prgeor Could you please be more specific?  I am not sure what to do with just bring this checking to line 249. Thanks

Reply:
```
if os.path.exists(PREVIOUS_REBOOT_CAUSE_FILE) or os.path.islink(PREVIOUS_REBOOT_CAUSE_FILE)
    os.remove(PREVIOUS_REBOOT_CAUSE_FILE)
 
# Create a symbolic link to previous-reboot-cause.json file
 os.symlink(REBOOT_CAUSE_HISTORY_FILE, PREVIOUS_REBOOT_CAUSE_FILE)
```